src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/CommentsInterceptor.kt

```diff
@@ -13,68 +14,112 @@ import okhttp3.Response
+        @Suppress("DEPRECATION")
+        val heading = StaticLayout(
+            "章末吐槽：",
+            paintHeading,
+            (WIDTH - 2 * X_PADDING).toInt(),
+            Layout.Alignment.ALIGN_NORMAL,
+            SPACING_MULT,
+            SPACING_ADD,
+            true,
+        )
+
+        val comments = parseChapterComments(response).toMutableList()
+        val paintBody = TextPaint().apply {
+            color = Color.BLACK
+            textSize = BODY_FONT_SIZE
+            typeface = Typeface.DEFAULT
+            isAntiAlias = true
+        }
+
+        var currentHeight = Y_PADDING + heading.height
+        val bodyLayouts = mutableListOf<StaticLayout>()
+        for (comment in comments) {
             @Suppress("DEPRECATION")
-            StaticLayout(it, paint, WIDTH - 2 * UNIT, Layout.Alignment.ALIGN_NORMAL, 1f, 0f, false)
-        }.takeWhile {
-            val lineHeight = it.height + UNIT
-            if (height + lineHeight <= MAX_HEIGHT) {
-                height += lineHeight
-                true
-            } else {
-                false
+            val layout = StaticLayout(
+                comment,
+                paintBody,
+                (WIDTH - 2 * X_PADDING).toInt(),
+                Layout.Alignment.ALIGN_NORMAL,
+                SPACING_MULT,
+                SPACING_ADD,
+                true,
+            )
+            val lineHeight = SPACING + layout.height
+            // If adding this comment doesn't exceed the max height, add it.
+            // If it does exceed and it's a single line, stop adding more comments.
+            // Otherwise, if it's multi-line, just skip it and try the next comment.
+            if (currentHeight + lineHeight <= MAX_HEIGHT) {
+                bodyLayouts.add(layout)
+                currentHeight += lineHeight
+            } else if (layout.lineCount == 1) {
+                break
             }
         }
 
-        val bitmap = Bitmap.createBitmap(WIDTH, height, Bitmap.Config.ARGB_8888)
-        bitmap.eraseColor(Color.WHITE)
-        val canvas = Canvas(bitmap)
+        // The bitmap height must be no more than MAX_HEIGHT
+        // and no less than its width to prevent automatic double-page splitting.
+        val bitmapHeight = (currentHeight + Y_PADDING).toInt().coerceIn(WIDTH, MAX_HEIGHT)
 
-        var y = UNIT
-        for (layout in layouts) {
-            canvas.save()
-            canvas.translate(UNIT_F, y.toFloat())
-            layout.draw(canvas)
-            canvas.restore()
-            y += layout.height + UNIT
+        val bitmap = Bitmap.createBitmap(WIDTH, bitmapHeight, Bitmap.Config.ARGB_8888)
+        Canvas(bitmap).apply {
+            drawColor(Color.WHITE)
+            heading.draw(this, X_PADDING, Y_PADDING)
+            var y = Y_PADDING + heading.height + SPACING
+            for (layout in bodyLayouts) {
+                layout.draw(this, X_PADDING, y)
+                y += layout.height + SPACING
+            }
```

Comment:
I think you can use fold instead, to avoid `var`:

```suggestion
            bodyLayouts.fold(Y_PADDING + heading.height + SPACING) { y, layout ->
                layout.draw(this, X_PADDING, y)
                y + layout.height + SPACING
            }
```

Reply:
IMO the current code is clearer.